tavern/util/loader.py

```diff
@@ -82,13 +90,54 @@ def construct_include(loader, node):
     if extension not in ('yaml', 'yml'):
         raise BadSchemaError("Unknown filetype '{}'".format(filename))
 
+    return filename
+
+
+def construct_include(loader, node):
+    """Include file referenced at node."""
+
+    filename = _get_yaml_filename(loader, node)
     with open(filename, 'r') as f:
         return yaml.load(f, IncludeLoader)
 
+def construct_load(loader, node):
+    """ Load test sections from a referenced file.
+
+    These sections are only loaded; they are not constructed as part of the
+    loaded YAML document. They are available to be constructed on demand, e.g.
+    using `!stage`.
+    """
+
+    filename = _get_yaml_filename(loader, node)
+    with open(filename, 'r') as f:
+        doc = yaml.load(f, IncludeLoader)
+        if "stages" in doc:
+            for name, stage in doc["stages"].items():
+                logger.info("Loading stage: %s", name)
+                logger.debug("Loading stage [%s]: %s", name, stage)
+                loader.loaded_stages[name] = stage
+
+    return None
+
+def construct_stage(loader, node):
+    """ Construct a stage from previously loaded stage.
+    """
+    stage_name = str(loader.construct_scalar(node))
+    try:
+        stage = loader.loaded_stages[stage_name]
+    except KeyError as e:
+        logger.error("The following key does not exist: %s", e.args[0])
```

Comment:
Does this depend on the order that things are defined in the file? eg, does it work if you have this:

```yaml
stages:
  - !stage abc

loads:
  - extra_stages.yaml
```

Reply:
Order is definitely important, what you've listed will not work with the current implementation. I don't know how strongly you feel about making order not relevant; in every config I've used (YAML or not) if including of external resources is supported, the resources need to be included before they can be referenced.